crates/completion/src/completions/postfix.rs

```diff
@@ -30,11 +30,17 @@ pub(crate) fn complete_postfix(acc: &mut Completions, ctx: &CompletionContext) {
     let receiver_text =
         get_receiver_text(dot_receiver, ctx.dot_receiver_is_ambiguous_float_literal);
 
-    let receiver_ty = match ctx.sema.type_of_expr(&dot_receiver) {
+    let mut receiver_ty = match ctx.sema.type_of_expr(&dot_receiver) {
         Some(it) => it,
         None => return,
     };
 
+    let mut ref_removed = false;
+    if let Some(removed) = receiver_ty.remove_ref() {
+        receiver_ty = removed;
+        ref_removed = true;
+    }
```

Comment:
We can get rid of this bool by just saving the ref stripped type in a different named variable, also this only removes one ref but we want to ideally remove all refs. Unwrapping here is fine since we always have a value at the start. Something like this:
```suggestion
    let no_ref_receiver_ty = std::iter::successors(Some(receiver_ty.clone()), |ty| ty.remove_ref()).unwrap();
```